boot/seal_test.go

```diff
@@ -1215,6 +1256,171 @@ func (s *sealSuite) TestResealKeyToModeenvFallbackCmdline(c *C) {
 	})
 }
 
+func (s *sealSuite) TestRunModeBootChains(c *C) {
```

Comment:
I see, do have higher level tests then that test the case where we need to consider both old and new grub assets locations?  I saw only this one doing that but maybe I missed them

Reply:
Right, I think we are missing a test for `ResealKeyToModeenv` with both chain present at the same time. We do test both. But separately.

Reply:
I have modified `TestResealKeyToModeenvWithSystemFallback` to also cover cases where we upgrade or downgrade. This calls `ResealKeyToModeenv` which is higher level.

But that is starting to be complicated. Specially because of ordering and factorization of boot chains. There is lots of ways to express allowed boot chains. But we only test one, which is very hard to guess. I spend a day to make it work.